pkg/orchestration/wiring/wiringutil/knownshapes/scaling.go

```diff
@@ -0,0 +1,61 @@
+package knownshapes
+
+import (
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/libshapes"
+)
+
+const (
+	// allows to select set of Kubernetes objects by list of scaling
+	SetOfScalingShape wiringplugin.ShapeName = "voyager.atl-paas.net/SetOfScalingShape"
+)
+
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin.Shape
+type SetOfScaling struct {
+	wiringplugin.ShapeMeta `json:",inline"`
+	Data                   SetOfScalingData `json:"data"`
+}
+
+// +k8s:deepcopy-gen=true
+type Scaling struct {
+	MinReplicas int32 `json:"minReplicas,omitempty"`
+	MaxReplicas int32 `json:"maxReplicas,omitempty"`
+	CPULimit    int32 `json:"cpuLimit,omitempty"`
+	MemoryLimit int32 `json:"memoryLimit,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type SetOfScalingData struct {
+	DeploymentResourceName smith_v1.ResourceName `json:"deploymentResourceName"`
+	Scaling                Scaling               `json:"scaling"`
+}
+
+func NewSetOfScaling(resourceName smith_v1.ResourceName, scaling Scaling) *SetOfScaling {
+	return &SetOfScaling{
+		ShapeMeta: wiringplugin.ShapeMeta{
+			ShapeName: SetOfScalingShape,
+		},
+		Data: SetOfScalingData{
+			DeploymentResourceName: resourceName,
+			Scaling:                scaling,
+		},
+	}
+}
+
+func (s *SetOfScaling) Name() wiringplugin.ShapeName {
```

Comment:
This method can be remove - it is implemented by the embedded `wiringplugin.ShapeMeta`.